webapp/first_snap/content/rust/package.yaml

```diff
@@ -0,0 +1,62 @@
+    fast and composable:
+    - Simple tasks should be easy.
+    - Performance trade offs should be exposed in the CLI interface.
+    - Composition should not come at the expense of performance.
+explain:
+  metadata:
+    - text: The name must be unique in the Snap Store. Valid snap names consist of lower-case alphanumeric characters and hyphens. They cannot be all numbers. They also cannot start or end with a hyphen.
+    - code: |
+        name: xsv
+    - warning: |
+        <span class="p-notification__status">Snap names are globally unique.</span> You should change any instance of <code>xsv</code> to <code>test-xsv-{name}</code> (where <code>{name}</code> is your name) before continuing to avoid conflicting with the official xsv snap.
+    - text: By specifying git for the version, the current git tag or commit will be used as the version string. Versions carry no semantic meaning in snaps.
+    - code: |
+        version: git
+    - text: The summary can not exceed 79 characters. You can use a pipe ‘|’ in the description key to declare a multi-line description.
+    - code: |
+        description: |
+  security:
+    - text: The next section describes the level of confinement applied to your app.
+    - code: |
+        confinement: devmode
+    - text: Snaps are containerised to ensure more predictable application behaviour and greater security.
+    - text: It’s best to start a snap with the confinement in warning mode, rather than strictly applied. This is indicated through the <code>devmode</code> keyword.
+    - text: Once an app is working well in devmode, you can review confinement violations, add appropriate interfaces, and switch to strict confinement.
+  base:
+    - text: In general, snaps cannot see the root filesystem on end user systems. This prevents conflict with other applications and increases security. However, applications still need some location to act as the root filesystem. They would also benefit from common libraries (e.g. libc) being in this root filesystem rather than bundled into each application.
+    - text: The base keyword specifies a special kind of snap that provides a minimal set of libraries common to most applications. It will be mounted as the root filesystem for your application.
+    - code: |
+        base: core18
+    - text: The <code>core18</code> base is recommended.
+  parts:
+    - text: |
+        Parts define what sources are needed to assemble your app. Parts can be anything: programs, libraries, or other needed assets. In this case we have one: the xsv source code. In other cases these can point to local directories, remote git repositories, or tarballs.
+    - code: |
+        parts:
+          xsv:
+            plugin: rust
+            source: .
+    - text: |
+        You can define the exact version of Rust with the optional <code>rust-revision</code> keyword. Dependencies from your <code>Cargo.toml</code> will also be bundled in the snap.
```

Comment:
>Dependencies from your <code>Cargo.toml</code> will also be bundled in the snap.

If you build and published a rust app outside of snaps, would the dependencies from cargo be included in your binary?

Reply:
This issue still needs to be addressed. As I understand it, cargo dependencies are statically linked.

* https://stackoverflow.com/a/29008355
* https://doc.rust-lang.org/cargo/guide/dependencies.html
* https://doc.rust-lang.org/reference/linkage.html
* https://www.reddit.com/r/rust/comments/8s8tv7/dynamic_linking_in_rust/